src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEModuleSymbol.vb

```diff
@@ -423,30 +423,44 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
         ''' <remarks>
-        ''' The returned assembly may also forward the type.
+        ''' The returned assemblies may also forward the type.
         ''' </remarks>
-        Friend Function GetAssemblyForForwardedType(ByRef fullName As MetadataTypeName, ignoreCase As Boolean, <Out> ByRef matchedName As String) As AssemblySymbol
-            Dim assemblyRef As AssemblyReferenceHandle = Me.Module.GetAssemblyForForwardedType(fullName.FullName, ignoreCase, matchedName)
-            Return If(assemblyRef.IsNil, Nothing, GetReferencedAssemblySymbol(assemblyRef))
+        Friend Function GetAssembliesForForwardedType(ByRef fullName As MetadataTypeName, ignoreCase As Boolean, <Out> ByRef matchedName As String) As (FirstSymbol As AssemblySymbol, SecondSymbol As AssemblySymbol)
+            Dim indices = Me.Module.GetAssemblyRefsForForwardedType(fullName.FullName, ignoreCase, matchedName)
+            
+            Dim firstSymbol = If(indices.FirstIndex < 0, Nothing, GetReferencedAssemblySymbol(indices.FirstIndex))
+            Dim secondSymbol = If(indices.SecondIndex < 0, Nothing, GetReferencedAssemblySymbol(indices.SecondIndex))
+
+            return (firstSymbol, secondSymbol)
         End Function
 
         Friend Iterator Function GetForwardedTypes() As IEnumerable(Of NamedTypeSymbol)
-            For Each forwarder As KeyValuePair(Of String, AssemblyReferenceHandle) In Me.Module.GetForwardedTypes()
-                Dim assembly As AssemblySymbol = GetReferencedAssemblySymbol(forwarder.Value)
-                If assembly Is Nothing Then
+            For Each forwarder As KeyValuePair(Of String, (FirstIndex As Integer, SecondIndex As Integer)) In Me.Module.GetForwardedTypes()
+                If forwarder.Value.FirstIndex < 0 Then
                     Continue For
                 End If
 
                 Dim name = MetadataTypeName.FromFullName(forwarder.Key)
-                Yield assembly.LookupTopLevelMetadataType(name, digThroughForwardedTypes:=True)
+
+                Dim firstSymbol = GetReferencedAssemblySymbol(forwarder.Value.FirstIndex)
+                Debug.Assert(DirectCast(firstSymbol, Object) IsNot Nothing, "Invalid indexes (out of bound) are discarded during reading metadata in PEModule.EnsureForwardTypeToAssemblyMap()")
+
+                If forwarder.Value.SecondIndex >= 0 Then
+                    Dim secondSymbol = GetReferencedAssemblySymbol(forwarder.Value.SecondIndex)
+                    Debug.Assert(DirectCast(secondSymbol, Object) IsNot Nothing, "Invalid indexes (out of bound) are discarded during reading metadata in PEModule.EnsureForwardTypeToAssemblyMap()")
+
+                    Dim forwardingErrorInfo = New DiagnosticInfo(MessageProvider.Instance, ERRID.ERR_TypeForwardedToMultipleAssemblies, forwarder.Key, firstSymbol.Name, secondSymbol.Name)
+                    Yield New MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo(Me, name, forwardingErrorInfo)
```

Comment:
Consider extracting a helper method to construct the `MissingMetadataTypeSymbol`, for use here and in `PEAssemblySymbol` and `SourceAssemblySymbol`.